src/gui/torrentcontenttreeview.cpp

```diff
@@ -107,6 +108,78 @@ void TorrentContentTreeView::keyPressEvent(QKeyEvent *event)
     }
 }
 
+void TorrentContentTreeView::setupDownloadPriorityMenu(QMenu *menu, const bool createSubMenu) const
+{
+    const auto model = qobject_cast<TorrentContentFilterModel *>(this->model());
+    Q_ASSERT(model);
+
+    const auto getSelectedRows = [this]() { return selectionModel()->selectedRows(TorrentContentModelItem::COL_PRIO); };
+
+    const auto applyPriorities = [&](const BitTorrent::DownloadPriority priority)
+    {
+        return [model, getSelectedRows, priority]()
+        {
+            for (const QModelIndex &index : asConst(getSelectedRows()))
+            {
+                model->setData(index, static_cast<int>(priority));
+            }
+            emit model->filteredFilesChanged();
+        };
+    };
+    const auto applyPrioritiesByOrder = [model, getSelectedRows]()
+    {
+        // Equally distribute the selected items into groups and for each group assign
+        // a download priority that will apply to each item. The number of groups depends on how
+        // many "download priority" are available to be assigned
+
+        const QModelIndexList selectedRows = getSelectedRows();
+        const qsizetype priorityGroups = 3;
+        const auto priorityGroupSize = std::max<qsizetype>(selectedRows.length() / priorityGroups, 1);
+
+        for (qsizetype i = 0; i < selectedRows.length(); ++i)
+        {
+            auto priority = BitTorrent::DownloadPriority::Ignored;
+            switch (i / priorityGroupSize)
+            {
+            case 0:
+                priority = BitTorrent::DownloadPriority::Maximum;
+                break;
+            case 1:
+                priority = BitTorrent::DownloadPriority::High;
+                break;
+            default:
+            case 2:
+                priority = BitTorrent::DownloadPriority::Normal;
+                break;
+            }
+
+            model->setData(selectedRows[i], static_cast<int>(priority));
+        }
+
+        emit model->filteredFilesChanged();
+    };
+
+    if (createSubMenu)
+    {
+        menu = menu->addMenu(tr("Priority"));
```

Comment:
I would use another variable `submenu` instead of reusing the same.

Reply:
Fixed by b96f28e19e464c3189263c63e6356e15e67b1cfe